packages/shopping/src/index.ts

```diff
@@ -5,12 +5,18 @@
 
 import {ShoppingApplication} from './application';
 import {ApplicationConfig} from '@loopback/core';
+import {RestBindings} from '@loopback/rest';
+import {addSecuritychema} from './utils/security-spec';
 export {ShoppingApplication, PackageInfo, PackageKey} from './application';
 
 export async function main(options?: ApplicationConfig) {
   const app = new ShoppingApplication(options);
 
   await app.boot();
+  let oaiSchema = app.getSync(RestBindings.API_SPEC);
+  addSecuritychema(oaiSchema);
+  console.log(oaiSchema.components!.securitySchemes);
+  app.bind(RestBindings.API_SPEC).to(oaiSchema);
```

Comment:
Can we use `app.api()` instead of this low-level implementation manipulating bindings?

See
https://github.com/strongloop/loopback-next/blob/70ad2ced8119ddf4b7da28ced5643edd4c5084de/packages/rest/src/rest.application.ts#L276-L289

Something along the following lines:

```ts
this.api({
  components: {
    ...BEARER_SECURITY_SCHEMA_SPEC,
  },
  security: {
    ...SECURITY_SPEC,
  }
});
```

Either way, the security schema should be configured from application constructor, not from the `main` function. It should be possible to start the application by calling `app.boot()` and `app.start()` directly (no `main` involved) and still have security schemas defined.

As a general rule, the `main` function should have a single responsibility - setup & start the application. Everything else should be handled by app's constructor or `boot`/`start`methods.